agent/functional_tests/util/utils.go

```diff
@@ -719,6 +730,20 @@ func RequireRegions(t *testing.T, supportedRegions []string, region string) {
 	}
 }
 
+// RequireInstanceTypes skips the test if current instance type is not a supported instance type
+func RequireInstanceTypes(t *testing.T, supportedTypePrefixes []string) {
+	iid, _ := ec2.NewEC2MetadataClient(nil).InstanceIdentityDocument()
```

Comment:
are we ok with adding this dependency? we were heading towards removing these, weren't we?

Reply:
since right now we are running functional tests across instance types, and the test can only pass on some instance types, this is the only way i know to avoid test failing. we can revisit this when our test facility support us to remove iid i think?